shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -556,18 +562,46 @@ public void onDisplayPlatformView(int viewId, int x, int y, int width, int heigh
   }
 
   public void onDisplayOverlaySurface(int id, int x, int y, int width, int height) {
+    FlutterView flutterView = (FlutterView) this.flutterView;
     if (!flutterViewConvertedToImageView) {
-      ((FlutterView) flutterView).convertToImageView();
+      flutterView.convertToImageView();
       flutterViewConvertedToImageView = true;
     }
+
+    FlutterImageView overlayView = overlayLayerViews.get(id);
+    if (overlayView.getParent() == null) {
+      flutterView.addView(overlayView);
+    }
+
+    FrameLayout.LayoutParams layoutParams = new FrameLayout.LayoutParams((int) width, (int) height);
+    layoutParams.leftMargin = (int) x;
+    layoutParams.topMargin = (int) y;
+    overlayView.setLayoutParams(layoutParams);
+    overlayView.setVisibility(View.VISIBLE);
+    overlayView.bringToFront();
+
+    currentFrameUsedOverlayLayerIds.add(id);
   }
 
   public void onBeginFrame() {
-    // TODO: Implement this method. https://github.com/flutter/flutter/issues/58288
+    currentFrameUsedOverlayLayerIds.clear();
   }
 
   public void onEndFrame() {
-    // TODO: Implement this method. https://github.com/flutter/flutter/issues/58288
+    for (int i = 0; i < overlayLayerViews.size(); i++) {
+      int key = overlayLayerViews.keyAt(i);
+      FlutterImageView overlayView = overlayLayerViews.valueAt(i);
+      if (currentFrameUsedOverlayLayerIds.contains(key)) {
+        overlayView.acquireLatestImage();
```

Comment:
The Android team to acquire the image off the UI thread.  This can be done later, just fyi.

Reply:
Same for the one above, but I guess that could go in a single PR.